app/scripts/controllers/workflowfileviewer.js

```diff
@@ -66,6 +86,81 @@ angular.module('dockstore.ui')
           function(result){
             $scope.selVersionName = $scope.successContent[0].version;
             $scope.selDescriptorName = $scope.successContent[0].descriptor;
+            $scope.fileContent = $scope.successContent[0].content;
+            var result = $scope.fileContent.toLowerCase();
+            m = [];
+            v = false;
+            if($scope.selDescriptorName === "cwl"){
```

Comment:
Same here. 

For a next step, we should sit down together and look at how we can create new web resources, one for workflows and one for tools, which can check whether an entry is valid. That way, you'll be able to re-use code and we won't need to duplicate this code between the UI and the launcher.